simpegEM1D/Waveforms.py

```diff
@@ -233,6 +235,92 @@ def piecewise_ramp(step_func, t_off, t_currents, currents, n=20, eps=1e-10):
             ) * const
```

Comment:
Yes, there are no loops in the _fast methods.  Note that the _diff_fast replaces the original call to piecewise_ramp twice. Rather than calling the same function twice, and re-computing the same stuff, i just unrolled the two calls, figured out the common stuff, created this new function